plugins/multilanguage.ts

```diff
@@ -176,7 +193,65 @@ export default function multilanguage(userOptions: Options) {
           document.head.appendChild(meta);
           document.head.appendChild(document.createTextNode("\n"));
         }
+
+        if (page.data.unmatchedLangUrl) {
+          appendHreflang("x-default", page.data.unmatchedLangUrl, document);
+        }
       }
     });
   };
 }
+
+function getUnmatchedLangPath(
+  currentPage: Page<Data>,
+  filteredPages: Page<Data>[],
+): string | undefined {
+  const { unmatchedLangUrl } = currentPage.data;
+
+  if (!unmatchedLangUrl) return void 0;
+
+  // If unmatchedLang is an external URL string
+  if (unmatchedLangUrl.startsWith("http")) {
+    return unmatchedLangUrl;
+  }
+
+  // If unmatchedLang is an source path string
+  if (unmatchedLangUrl.startsWith("/")) {
+    const regUrl = new RegExp(`^\/${trim(unmatchedLangUrl, "\/")}\/?`, "i");
+    const langSelectorPage = filteredPages.find(({ data }) => {
+      const [isMatch] = data.url.match(regUrl) ?? [];
+      return !!isMatch;
+    })?.data;
```

Comment:
Why do you use a regular expression here?
Why don't simply:

```js
const exists = filteredPages.some(({ data }) => data.url === unmatchedLangUrl);
```

Reply:
If you go for absolute comparision, it will break when `url: /abc/` and `unmatchedLangUrl: /abc`.

Reply:
In this case, I would show the warning of "page not found", because technically they are different urls.